app/javascript/src/utils/compiler.js

```
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
+        result = i
+        break
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * @param {string} source
+ * @returns {string}
+ */
+function evaluateConditionals(source) {
+  const conditionalStartRegex = /@if *\( *((?:.|\n)+?) *(==|!=) *((?:.|\n)+?)\) *[ \n]*\{/g
+  const conditionalElseStartRegex = / *@else *\{/
+
+  let match
+  while ((match = conditionalStartRegex.exec(source)) != null) {
+    const [matchedConditionalStartText, left, operation, right] = match
+    const afterMatchedTextIndex = match.index + matchedConditionalStartText.length
+
+    const matchingClosingBracketIndex = findMatchingClosingBracketIndex(source, afterMatchedTextIndex)
+    if (matchingClosingBracketIndex < 0) {
+      // TODO: warn user?
+      continue
+    }
+
+    let conditionalEndingIndex = matchingClosingBracketIndex
+
+    const trueBlockContent = source.substring(afterMatchedTextIndex, matchingClosingBracketIndex)
+    let falseBlockContent = ""
+
+    conditionalElseStartRegex.lastIndex = matchingClosingBracketIndex
+    const elseMatch = conditionalElseStartRegex.exec(source)
+    if (elseMatch != null) {
+      const afterElseMatchedTextIndex = elseMatch.index + elseMatch[0].length
+      const matchingClosingBracketForElseIndex = findMatchingClosingBracketIndex(source, afterElseMatchedTextIndex)
+      if (matchingClosingBracketForElseIndex > 0) {
+        falseBlockContent = source.substring(afterElseMatchedTextIndex, matchingClosingBracketForElseIndex)
+        conditionalEndingIndex = matchingClosingBracketForElseIndex
+      } // TODO: else warn user?
+    }
+
+    let passed = null
+    const sanitizedLeft = left.trimEnd()
+    const sanitizedRight = right.trimStart()
+    switch (operation) {
+      case "==": {
+        passed = sanitizedLeft === sanitizedRight
+        break
+      }
+      case "!=": {
+        passed = sanitizedLeft !== sanitizedRight
+        break
+      }
+      // TODO: contains operator, regex operator?
```

For these I was thinking
```
abc *= b     # compare using `String,includes`: `lefthand.includes(righthand)`
abc ~= /B/i  # parse regex and compare using `RegExp.test()`: `parsedRighthandRegex.test(lefthand)`
```

---

In the case of the regex operator, what should happen if the righthand is not a valid regex? Make the operator silently return false? Warn to the user somehow?

---

Yep, I think in that case it should be silent. Any warnings should be in the linter.